signer/src/daemon/engine.c

```diff
@@ -709,19 +669,9 @@ engine_update_zones(engine_type* engine, ods_status zl_changed)
         warnings += dnsconfig_zone(engine, zone);
 
         if (zone->zl_status == ZONE_ZL_ADDED) {
-            ods_log_assert(task);
-            pthread_mutex_lock(&zone->zone_lock);
-            zone->task = task;
-            pthread_mutex_unlock(&zone->zone_lock);
-            /* TODO: task is reachable from other threads by means of
-             * zone->task. To fix this we need to nest the locks. But
-             * first investigate any possible deadlocks. */
-            status = schedule_task(engine->taskq, task, 0, 0);
+            schedule_scheduletask(engine->taskq, TASK_SIGNCONF, zone->name, zone, &zone->zone_lock, 0);
         } else if (zl_changed == ODS_STATUS_OK) {
-            /* always try to update signconf */
-            pthread_mutex_lock(&zone->zone_lock);
-            status = zone_reschedule_task(zone, engine->taskq, TASK_SIGNCONF);
-            pthread_mutex_unlock(&zone->zone_lock);
+            schedule_scheduletask(engine->taskq, TASK_FORCESIGNCONF, zone->name, zone, &zone->zone_lock, 0);
         }
         if (status != ODS_STATUS_OK) {
```

Comment:
This code is no longer reachable. Which is a shame schedule_scheduletask() could easily return a status.